### PR TITLE
Explore: Download traces as JSON in Explore Inspector

### LINKED ISSUE #27852: Download and Upload traces in Explore tab
**What would you like to be added** / **Why is this needed**
I would like to be able to download trace from Explorer view as a JSON file. This is possible in Jaeger, and is mostly used for archiving specific traces, or for further analysis via custom tools.

Once downloaded, it would also be nice to have an ability to upload trace back for viewing it in the UI again :)

### PR DESCRIPTION
**Pull request by connorlindsey**
**What this PR does / why we need it**:
This PR adds the ability to download traces as JSON in their relevant format (Jaeger, Zipkin, or OTLP). 

**Which issue(s) this PR fixes**:

Fixes #27852

**Special notes for your reviewer**:

TODO:
- [x] Jaeger transform and tests
- [x] Zipkin transform and tests
- [x] OTLP transform and tests
- [x] Download file in Inspect Data tab
- [x] Documentation

### CODE REVIEW

File: public/app/plugins/datasource/tempo/datasource.ts [hunk 1]
```diff
@@ -84,7 +84,10 @@ export class TempoDatasource extends DataSourceWithBackend<TempoQuery, TempoJson
                 .map((field) => field.matcherRegex) || [];
             if (!traceLinkMatcher || traceLinkMatcher.length === 0) {
               return throwError(
-                'No Loki datasource configured for search. Set up Derived Fields for traces in a Loki datasource settings and link it to this Tempo datasource.'
```
Comment (connorlindsey) [inline public/app/plugins/datasource/tempo/datasource.ts:87]: Updated deprecated RxJS method

File: docs/sources/explore/explore-inspector.md [hunk 2]
```diff
@@ -57,6 +57,14 @@ Grafana generates a TXT file in your default browser download location. You can
 1. Inspect the log query results as described above.
 1. Click **Download logs**.
 
+### Download trace results
+
+Grafana generates a JSON file for the trace results based on the trace datasource type. Jaeger, Zipkin, and OTLP formats are supported.
```
Comment (achatterjee-grafana) [inline docs/sources/explore/explore-inspector.md:62]:
```suggestion
Based on the data source type, Grafana generates a JSON file for the trace results in one of the supported formats: Jaeger, Zipkin, or OTLP formats.
```

File: docs/sources/explore/explore-inspector.md [hunk 3]
```diff
@@ -57,6 +57,14 @@ Grafana generates a TXT file in your default browser download location. You can
...
+
+1. Open the inspector.
+1. Inspect the trace query results as described above.
```
Comment (achatterjee-grafana) [inline docs/sources/explore/explore-inspector.md:65]: Link to #inspect-raw-query-results.

**Review by achatterjee-grafana — COMMENTED** [review]
Added doc review.

**Review by zoltanbedi — COMMENTED** [review]
Nice!

- Zipkin looks good
- Jaeger looks good
- Tempo looks different
Screenshot 2021-09-02 at 12 26 51

File: public/app/plugins/datasource/tempo/resultTransformer.ts [hunk 4]
```diff
@@ -10,8 +10,9 @@ import {
   TraceLog,
   TraceSpanRow,
 } from '@grafana/data';
-import { SpanKind, SpanStatusCode } from '@opentelemetry/api';
+import { SpanKind, SpanStatus, SpanStatusCode } from '@opentelemetry/api';
 import { collectorTypes } from '@opentelemetry/exporter-collector';
+import { opentelemetryProto } from '@opentelemetry/exporter-collector/build/src/types';
```
[the remark this answers is not in the document]
Comment (connorlindsey) [reply public/app/plugins/datasource/tempo/resultTransformer.ts:15]: Yeah, you're right. Must've been auto-import. Updated that and will include it when I push any fixes that resolve Tempo not uploading correctly.

**Comment by connorlindsey** [comment]
@zoltanbedi Got Tempo working although the files don't match perfectly. The main difference is that the file you uploaded batches by service name and some other criteria (because there are multiple batches for some services like app and loadgen). I batched just by service name, so some of the batches got combined. This gives the same result in the UI and I think is close enough.

**Review by zoltanbedi — APPROVED** [review]
Looks good to me.

One thing that I noticed that is not related to this pr is that setting the spankind in getSpanTags function is wrong.
Screenshot 2021-09-08 at 13 11 03

Screenshot 2021-09-08 at 13 07 56

**Comment by connorlindsey** [comment]
That makes sense, I thought I had noticed some weirdness around that. I'll make a note to fix that.

**Pull request merged**